### ISSUE TITLE
uniqueIdentifier property and multiple Entities 

### ISSUE THREAD

**Issue by JDSX** [oskarirauta/EnsembleExample#1]
Your project really helped out for us swift developers, but I was left wondering about uniques Identifier property. When are you actually setting the unique ID Property ? is that don't automatically via the following function; 

```
func persistentStoreEnsemble(ensemble: CDEPersistentStoreEnsemble!, globalIdentifiersForManagedObjects objects: [AnyObject]!) -> [AnyObject]! {
    return (objects as NSArray).valueForKeyPath("uniqueIdentifier") as! [AnyObject]
}
```

Also if you had multiple entities I am correct thinking the function will just have an array of all the `NSManagedObjects` regardless of the entity name and use their uniqueIdentifiers ? 

Because I am working with an existing project and was thinking about creating ID's for existing objects; But if I am correct all I need to do is add the uniqueIdentifier property to the entities and the function will take care of the rest ?

Just a little confused coming from Drews example, 

```
func persistentStoreEnsemble(ensemble: CDEPersistentStoreEnsemble, globalIdentifiersForManagedObjects objects: [NSManagedObject]) -> [NSObject] {
    let numberHolders = objects as! [NumberHolder]
    return numberHolders.map { $0.uniqueIdentifier }
}
```

**Comment by oskarirauta** [comment]
> On 9.4.2016, at 20:59, jsingh notifications@github.com wrote:
> 
> Your project really helped out for us swift developers, but I was left wondering about uniques Identifier property. When are you actually setting the unique ID Property ? is that don't automatically via the following function;
> 
> func persistentStoreEnsemble(ensemble: CDEPersistentStoreEnsemble!, globalIdentifiersForManagedObjects objects: [AnyObject]!) -> [AnyObject]! {
>     return (objects as NSArray).valueForKeyPath("uniqueIdentifier") as! [AnyObject]
> }
> Also if you had multiple entities I am correct thinking the function will just have an array of all the NSManagedObjects regardless of the entity name and use their uniqueIdentifiers ?
> 
> Because I am working with an existing project and was thinking about creating ID's for existing objects; But if I am correct all I need to do is add the uniqueIdentifier property to the entities and the function will take care of the rest ?
> 
> Just a little confused coming from Drews example,
> 
> func persistentStoreEnsemble(ensemble: CDEPersistentStoreEnsemble, globalIdentifiersForManagedObjects objects: [NSManagedObject]) -> [NSObject] {
>     let numberHolders = objects as! [NumberHolder]
>     return numberHolders.map { $0.uniqueIdentifier }
> }

Yes. It should be that simple that you just add uniqueIdentifier to entity name. Can’t remember this anymore, but feel free to fork and make tests and then you should make a pull request to this if changes are necessary :)
